main.tf

```diff
@@ -117,10 +117,23 @@ resource "google_project_iam_binding" "filestore_backup_runner_file_editor" {
 
   condition {
     title      = "${google_filestore_instance.default.name} instance"
-    expression = "resource.name.startsWith('projects/${local.project_id}/locations/${local.region}/backups/${google_filestore_instance.default.name}')"
+    expression = format(
+      "resource.name.startsWith('projects/%s/locations/%s/backups/%s')",
+      local.project_id,
+      local.region,
+      google_filestore_instance.default.name
+    )
   }
 }
 
+# Extra permissions for listing backups only 
```

Comment:
The `roles/file.editor` already contains `file.backups.list` permission:
```
$ gcloud iam roles describe roles/file.editor --format="flattened(includedPermissions)" | grep file.backups.list
includedPermissions[5]:  file.backups.list
includedPermissions[6]:  file.backups.listEffectiveTags
includedPermissions[7]:  file.backups.listTagBindings
```
Ergo, the cloud function service account should have the necessary permission 🤔 The issue is with IAM condition – it targets specific backup (last `%s`):
```hcl
 "resource.name.startsWith('projects/%s/locations/%s/backups/%s')",
```
I think relaxing IAM condition should make a trick, while it slightly violates the principle of least privilege:
```hcl
 "resource.name.startsWith('projects/%s/locations/%s/backups)",
```

Reply:
As I said in the description.  the problem is because the `list` function does not play nicely with the condition part of the IAM.  The only way I could get it to work with the `editor` role only is if I removed the condition completely.